### ISSUE TITLE
Uses deprecated client_golang API

### ISSUE THREAD

**Issue by TheTincho** [prometheus/pushgateway#166]
Hi,

The pushgateway has an old version of the client_golang vendored, and uses deprecated API methods that have now been removed from the library:

```
src/github.com/prometheus/pushgateway/main.go:67:2: undefined: prometheus.SetMetricFamilyInjectionHook
```

Please, can you update the vendoring and replace that call with the current API?

**Comment by SuperQ** [comment]
I think this is related/duplicates https://github.com/prometheus/pushgateway/issues/149

**Comment by beorn7** [comment]
This is very easy to fix (but not related to #149). I thought I had already done so.
